NBXplorer/Controllers/MainController.PSBT.cs

```diff
@@ -323,7 +323,15 @@ public partial class MainController
 				if (request.SpendAllMatchingOutpoints is true)
 					txBuilder.SendAllRemainingToChange();
 				psbt = txBuilder.BuildPSBT(false);
-				hasChange = psbt.Outputs.Any(o => o.ScriptPubKey == change.ScriptPubKey);
+				hasChange =  psbt.Outputs.Any(o => o.ScriptPubKey == change.ScriptPubKey);
+				if (request.DonateChangeToMiners && hasChange)
+				{
+					var txTmp = psbt.GetGlobalTransaction();
+					txTmp.Outputs.RemoveAll(o => o.ScriptPubKey == change.ScriptPubKey);
+					psbt = txBuilder.CreatePSBTFrom(txTmp, false);
+					hasChange =  psbt.Outputs.Any(o => o.ScriptPubKey == change.ScriptPubKey);
```

Comment:
`hasChange = false` no ?

Reply:
You always have to define a change address to the tx builder if there is not an exact spend, so has change is almost always true unless the requests is specifically crafted. I then remove the change output manually